hpx/traits.hpp

```diff
@@ -104,6 +104,9 @@ namespace hpx { namespace traits
     struct default_chunk_size;
 
     ///////////////////////////////////////////////////////////////////////////
+    template <typename Range>
+    struct is_range;
```

Comment:
I'd suggest making the `is_range` trait SFINAE friendly by giving it an `typename Enable = void` argument, similar to all the other traits.